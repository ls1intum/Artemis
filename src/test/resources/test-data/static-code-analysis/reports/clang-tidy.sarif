{
  "runs": [
    {
      "results": [
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "/var/tmp/testing-dir/assignment/src/main.cpp"
                },
                "region": {
                  "startColumn": 16,
                  "startLine": 5
                }
              }
            }
          ],
          "message": {
            "text": "Dereference of null pointer (loaded from variable 'p')"
          },
          "ruleId": "clang-analyzer-core.NullDereference"
        }
      ],
      "tool": {
        "driver": {
          "name": "clang-tidy"
        }
      }
    }
  ],
  "version": "2.1.0"
}
