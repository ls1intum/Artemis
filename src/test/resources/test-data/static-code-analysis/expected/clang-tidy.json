{
  "tool": "CLANG_TIDY",
  "issues": [
    {
      "filePath": "/var/tmp/testing-dir/assignment/src/main.cpp",
      "startLine": 5,
      "endLine": 5,
      "startColumn": 16,
      "endColumn": 17,
      "rule": "clang-analyzer-core.NullDereference",
      "category": "Lint",
      "message": "Dereference of null pointer (loaded from variable 'p')",
      "priority": "warning"
    }
  ]
}
